src/components/Dashboard/TodoList.vue

```diff
@@ -1043,6 +1031,20 @@ export default {
       // if tasks changes, reload them
       // (does not fire on initial page load)
       this.loadData()
+    },
+    taskItems () {
```

Comment:
First, this will now fire every time a task is added to this array, which means it's more doing work than it needs to.
Second, why was emitting these values after the main list is processed (ie, where it was before) a problem? I don't understand the timing issue.

Reply:
What i found in testing, was the `loadData` was the only place the task `emit`'s were called which happens when the component was created. At point of that initial emit, the taskItems didn't seem to have the values yet, even tho the `loadFilingItem` was being `awaited`.  I could look a little deeper into the asynchronous behaviour here..

Reply:
Maybe there's something missing to make the "await loadFilingItem" work?

Reply:
Absolutely, i'll look deeper into this!

Reply:
The best explanation i could find...
 "Using Babel will transform async/await to generator function and using forEach means that each iteration has an individual generator function, which has nothing to do with the others. so they will be executed independently and has no context of next()"...

Long story short, await does not seem to work with the forEach function :P

Doing a little refactoring here and fixing some tests!